qiskit/_quantumcircuit.py

```diff
@@ -298,3 +300,46 @@ def qasm(self):
         for instruction in self.data:
             string_temp += instruction.qasm() + "\n"
         return string_temp
+
+    def draw(self, scale=0.7, filename=None, style=None, output='text',
+             interactive=False, line_length=None):
+        """Draw the quantum circuit
+
+        Using the output parameter you can specify the format. The choices are:
+        0. text: ASCII art string
+        1. latex: high-quality images, but heavy external software dependencies
+        2. matplotlib: purely in Python with no external dependencies
+
+        Defaults to an overcomplete basis, in order to not alter gates.
+
+        Args:
+            scale (float): scale of image to draw (shrink if < 1)
+            filename (str): file path to save image to
+            style (dict or str): dictionary of style or file name of style
+                                 file. You can refer to the
+                                 :ref:`Style Dict Doc <style-dict-doc>` for
+                                 more information on the contents.
+            output (str): Select the output method to use for drawing the
+                circuit. Valid choices are `text`, `latex`, `latex_source`,
+                `mpl`.
+            interactive (bool): when set true show the circuit in a new window
+                (cannot inline in Jupyter). Note when used with the
+                latex_source output type this has no effect
+            line_length (int): sets the length of the lines generated by `text`
+        Returns:
+            PIL.Image: an in-memory representation of the circuit diagram
+            String: (outputs `text` and `latex_source`). The ascii art or the
+                LaTeX source code.
+        Raises:
+            VisualizationError: when an invalid output method is selected
+
+        """
+        return visualization.circuit_drawer(self, scale=scale,
+                                            filename=filename, style=style,
+                                            output=output,
+                                            interactive=interactive,
+                                            line_length=line_length)
+
+    def __str__(self):
```

Comment:
should this be __repr__

Reply:
I kept it as string because my intent was to have it render the circuit text diagram when it was treated as a string, so when doing `print(circuit)` or `str(circuit)`. Doing `__repr__` would also mean if you just ran a line `circuit` it would print the circuit text diagram, which I didn't think we wanted. But it's simple enough to change if I'm wrong on that point.